Error on blsky login 500

On bluesky login issue 500, the bot dont catch the reject and the bot crash
- check error handler

---
- hook response: 502 bad gateway
- server log

```
[2024-05-23T20:20:00.017] [INFO] BlueSkyService - login botensky.bsky.social@https://bsky.social
[2024-05-23T20:20:10.503] [ERROR] BlueSkyService - XRPCError: Internal Server Error
    at ServiceClient.call (/opt/render/project/src/node_modules/@atproto/xrpc/dist/client.js:106:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async BskyAgent.login (/opt/render/project/src/node_modules/@atproto/api/dist/agent.js:214:25) {
  status: 500,
  error: 'InternalServerError',
  success: false,
  (...)
  [2024-05-23T20:20:10.504] [WARN] BotService - plugin response status:undefined msg:Failed to log in — double check your credentials and try again.
	node:_http_server:352
    throw new ERR_HTTP_INVALID_STATUS_CODE(originalStatusCode);
          ^
RangeError [ERR_HTTP_INVALID_STATUS_CODE]: Invalid status code: undefined
    at ServerResponse.writeHead (node:_http_server:352:11)
    at ServerResponse._implicitHeader (node:_http_server:338:8)
    at write_ (node:_http_outgoing:945:9)
    at ServerResponse.end (node:_http_outgoing:1056:5)
    at ServerResponse.send (/opt/render/project/src/node_modules/express/lib/response.js:233:10)
    at ServerResponse.json (/opt/render/project/src/node_modules/express/lib/response.js:279:15)
    at file:///opt/render/project/src/src/services/ExpressServer.js:85:58
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_HTTP_INVALID_STATUS_CODE'
```

--- a/src/plugins/Plantnet.js
+++ b/src/plugins/Plantnet.js
@@ -1,7 +1,6 @@
 import log4js from 'log4js';
-import {arrayIsNotEmpty, clone, loadJsonResource} from "../lib/Common.js";
+import {isSet, arrayIsNotEmpty, clone, loadJsonResource} from "../lib/Common.js";
 import {firstImageOf, postHtmlOf, postImageOf, postInfoOf, postLinkOf, postTextOf} from "../domain/post.js";
-import {isSet} from "node:util/types";
 
 const PLANTNET_MINIMAL_PERCENT = 20;
 const PLANTNET_MINIMAL_RATIO = PLANTNET_MINIMAL_PERCENT / 100;

--- a/src/services/BotService.js
+++ b/src/services/BotService.js
@@ -61,7 +61,7 @@ export default class BotService {
                     resolve(result);
                 })
                 .catch(err => {
-                    bot.logger.warn("plugin response status:" + err.status + " msg:" + err.message);
+                    bot.logger.warn(`plugin error: ${err.message}`);
                     bot.newsService.add(err.html ? err.html : err.message);
                     reject(err);
                 });

--- a/src/services/ExpressServer.js
+++ b/src/services/ExpressServer.js
@@ -8,7 +8,7 @@ import {
     cacheGetVersion
 } from "../lib/MemoryCache.js";
 import {unauthorized} from "../lib/CommonApi.js";
-import {generateErrorId} from "../lib/Common.js";
+import {isSet, generateErrorId} from "../lib/Common.js";
 import {StatusCodes} from "http-status-codes";
 
 const __dirname = path.resolve();
@@ -71,6 +71,12 @@ export default class ExpressServer {
         res.json({version});
     }
 
+    /**
+     * trigger bot action
+     * @param req
+     * @param res cronjob limits the response payload (#9) and read only 64KB
+     * @returns {Promise<void>}
+     */
     async hookResponse(req, res) {
         const expressServer = this;
         try {
@@ -82,7 +88,10 @@ export default class ExpressServer {
             if (doSimulate || doAction) {
                 expressServer.botService.process(remoteAdd, doSimulate, pluginName)
                     .then(() => res.status(200).json({success: true}))
-                    .catch(err => res.status(err.status).json({success: false, message: err}));
+                    .catch(err => {
+                        const status = isSet(err.status) ? err.status : 500;
+                        res.status(status).json({success: false, status}); // do not include details in response
+                    });
             } else {
                 this.logger.debug(StatusCodes.UNAUTHORIZED, JSON.stringify({code: 401, doSimulate, doAction}));
                 unauthorized(res, UNAUTHORIZED_FRIENDLY);

